### PR TITLE
fix: access grants when RBAC is disabled

### PR DESCRIPTION
**Pull request by tgmendes**
Prevents issues when orgs are not properly linked in WorkOS and RBAC is disabled.

## Summary
- bypass WorkOS membership and organization linkage lookups in `ListGrants` when RBAC is disabled
- return full-access grant payloads when grants are not being enforced
- add regression coverage for non-enterprise and RBAC-disabled access grant responses

### CODE REVIEW

File: server/internal/access/impl.go [hunk 1]
```diff
@@ -612,6 +612,21 @@ func (s *Service) ListGrants(ctx context.Context, _ *gen.ListGrantsPayload) (*ge
...
+	if ac.AccountType != "enterprise" {
+		return &gen.ListUserGrantsResult{Grants: fullAccessGrantPayloads()}, nil
+	}
+	enabled, err := s.access.features.IsFeatureEnabled(ctx, ac.ActiveOrganizationID, productfeatures.FeatureRBAC)
+	if err != nil {
+		return nil, oops.E(oops.CodeUnexpected, err, "check RBAC feature").Log(ctx, s.logger)
+	}
+	if !enabled {
+		return &gen.ListUserGrantsResult{Grants: fullAccessGrantPayloads()}, nil
+	}
```
[the remark this answers is not in the document]
Comment (tgmendes) [reply server/internal/access/impl.go]: Switched `ListGrants` to reuse `s.access.shouldEnforce(ctx)` so it stays aligned with the enforcement path rather than duplicating the checks here. Also added a regression test for the enterprise + `SessionID == nil` case to cover the mismatch you pointed out.

**Review by adaam2 — APPROVED** [review]
